### ISSUE TITLE
Accessibility - Material input text type phone cannot be selected with TalkBack

### ISSUE THREAD

**Issue by DiegoGW** [flutter/flutter#134975]
### Is there an existing issue for this?

- [X] I have searched the [existing issues](https://github.com/flutter/flutter/issues)
- [X] I have read the [guide to filing a bug](https://flutter.dev/docs/resources/bug-reports)

### Steps to reproduce

1. In Android turn on Talkback (Pixel 6 - Android 13)
2. Go to any app that includes text field widget provided in the code section (similar to https://gallery.flutter.dev/#/demo/text-field )
3. Go to material text fields
4. Try to select phone field

### Expected results

Phone field should be selectable

### Actual results

Phone field cannot be selected with Talkback.
This is not replicable in iPhone with VoiceOver.

### Code sample

<details><summary>Code sample</summary>

This is the widget I’m using in Flutter:
```
class MaterialTextFieldView extends StatelessWidget {
  const MaterialTextFieldView({super.key});

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(
        backgroundColor: Theme.of(context).colorScheme.inversePrimary,
        title: const Text('Text fields'),
      ),
      body: const SingleChildScrollView(
        padding: EdgeInsets.symmetric(
          horizontal: 24,
          vertical: 16,
        ),
        child: Column(
          mainAxisAlignment: MainAxisAlignment.center,
          children: <Widget>[
            TextField(
              decoration: InputDecoration(
                labelText: 'Left to right',
              ),
            ),
            SizedBox(height: 48),
            TextField(
              decoration: InputDecoration(
                labelText: 'Right to left',
              ),
              textDirection: TextDirection.rtl,
            ),
            SizedBox(height: 48),
            TextField(
              maxLines: 2,
              decoration: InputDecoration(labelText: 'Max lines (2)'),
              textDirection: TextDirection.ltr,
            ),
            SizedBox(height: 48),
            PasswordTextField(),
            SizedBox(height: 48),
            PhoneNumberTextField(),
            SizedBox(height: 48),
          ],
        ),
      ),
    );
  }
}

```

</details>

### Screenshots or Video

<details>
<summary>Screenshots / Video demonstration</summary>

https://drive.google.com/file/d/10G8wtou_PG4lcOBAi7xSZn0VL6eQQEHS/view?usp=sharing

<img width="255" alt="Screenshot 2023-09-18 at 7 17 25 PM" src="https://github.com/flutter/flutter/assets/11681325/d5814148-4008-4690-92b1-9f4e36284ac7">

</details>

### Logs

<details><summary>Logs</summary>

```console
[Paste your logs here]
```

</details>

### Flutter Doctor output

<details><summary>Doctor output</summary>

```console
[✓] Flutter (Channel master, 3.14.0-14.0.pre.332, on macOS 13.5 22G74 darwin-arm64, locale en-US)
    • Flutter version 3.14.0-14.0.pre.332 on channel master at /Users/diegogawenda/fvm/versions/master
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 1b18b1327a (2 days ago), 2023-09-17 04:07:02 -0400
    • Engine revision 57a4ff3c7f
    • Dart version 3.2.0 (build 3.2.0-171.0.dev)
    • DevTools version 2.28.0-dev.8

[✓] Android toolchain - develop for Android devices (Android SDK version 34.0.0)
    • Android SDK at /Users/diegogawenda/Library/Android/sdk
    • Platform android-34, build-tools 34.0.0
    • ANDROID_HOME = /Users/diegogawenda/Library/Android/sdk
    • ANDROID_SDK_ROOT = /Users/diegogawenda/Library/Android/sdk
    • Java binary at: /Applications/Android Studio.app/Contents/jbr/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 17.0.6+0-17.0.6b802.4-9586694)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 14.3.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Build 14E300c
    • CocoaPods version 1.12.1

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 2022.2)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 17.0.6+0-17.0.6b802.4-9586694)

[✓] VS Code (version 1.80.2)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.72.0

[✓] Connected device (3 available)
    • Pixel 6 (mobile) • 1A141FDF600823 • android-arm64  • Android 13 (API 33)
    • macOS (desktop)  • macos          • darwin-arm64   • macOS 13.5 22G74 darwin-arm64
    • Chrome (web)     • chrome         • web-javascript • Google Chrome 117.0.5938.88

[✓] Network resources
    • All expected network resources are available.
```

</details>

**Comment by huycozy** [comment]
I'm re-verifying this on the latest Fluter stable 3.29.2, on `Pixel 7, Android 15`:
The `PhoneNumberTextField` widget is undefined in the provided sample code. I checked this with minimal code below, but I see the field is selectable as expected with TalkBack.

Perhaps the TextField needs specific attributes to illustrate the case? 
@DiegoGW: Can you please provide a complete sample code?

<details open>
<summary>Sample code</summary>

```dart
TextField(
      keyboardType: TextInputType.number,
),
```
</details>

<details>
<summary>Demo video</summary>

https://github.com/user-attachments/assets/2e27f1ca-f673-4dfd-9bbb-d61f823d25b9

</details>

<details>
<summary>flutter doctor -v</summary>

```console
[✓] Flutter (Channel stable, 3.29.2, on macOS 15.3.2 24D81 darwin-arm64, locale en-US) [118ms]
    • Flutter version 3.29.2 on channel stable at /Users/huym4/Documents/WORKING/SDK/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision c236373904 (2 weeks ago), 2025-03-13 16:17:06 -0400
    • Engine revision 18b71d647a
    • Dart version 3.7.2
    • DevTools version 2.42.3

[✓] Android toolchain - develop for Android devices (Android SDK version 35.0.1) [1,206ms]
    • Android SDK at /Users/huym4/Library/Android/sdk
    • Platform android-35, build-tools 35.0.1
    • ANDROID_HOME = /Users/huym4/Library/Android/sdk
    • Java binary at: /Applications/Android Studio.app/Contents/jbr/Contents/Home/bin/java
      This is the JDK bundled with the latest Android Studio installation on this machine.
      To manually set the JDK path, use: `flutter config --jdk-dir="path/to/jdk"`.
    • Java version OpenJDK Runtime Environment (build 21.0.5+-13047016-b750.29)
    • All Android licenses accepted.

[✗] Xcode - develop for iOS and macOS [61ms]
    ✗ Xcode installation is incomplete; a full installation is necessary for iOS and macOS development.
      Download at: https://developer.apple.com/xcode/
      Or install Xcode via the App Store.
      Once installed, run:
        sudo xcode-select --switch /Applications/Xcode.app/Contents/Developer
        sudo xcodebuild -runFirstLaunch
    ✗ CocoaPods not installed.
        CocoaPods is a package manager for iOS or macOS platform code.
        Without CocoaPods, plugins will not work on iOS or macOS.
        For more info, see https://flutter.dev/to/platform-plugins
      For installation instructions, see https://guides.cocoapods.org/using/getting-started.html#installation

[✓] Chrome - develop for the web [25ms]
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 2024.3) [24ms]
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 21.0.5+-13047016-b750.29)

[✓] VS Code (version 1.98.2) [24ms]
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.106.0

[✓] Connected device (3 available) [164ms]
    • Pixel 7 (mobile) • 2B171FDH20084L • android-arm64  • Android 15 (API 35)
    • macOS (desktop)  • macos          • darwin-arm64   • macOS 15.3.2 24D81 darwin-arm64
    • Chrome (web)     • chrome         • web-javascript • Google Chrome 134.0.6998.166

[✓] Network resources [431ms]
    • All expected network resources are available.

! Doctor found issues in 1 category.

```

</details>
